ensembl-track-api.openapi.yaml

```diff
@@ -20,11 +20,52 @@ info:
   description: Backend service serving a list of available genome browser tracks and track categories.
   contact:
     email: andres@ebi.ac.uk
-  version: 0.0.1
+  version: 0.1.0
 servers:
   - url: http://beta.ensembl.org/api/tracks
     description: Production server
 paths:
+  /genomes:
```

Comment:
Following the payload structure of the other endpoints, we could use `/genomes` to get/post all tracks in a single payload (see [current data](https://github.com/Ensembl/ensembl-web-track-api/blob/track-api-json/data/track_categories.json)), but that would not be scalable (imagine the payload for thousands of genomes/millions of tracks), so I here I propose implementing track data submission in stages: first create genome ID entries (`/genomes`), then add (all or some) tracks per genome (`/track_categories/{genome_id}`).

Reply:
Again, the same mistake is being repeated here. The existing file containing the data is driving the decision of how the models would look like, what endpoint it would have and how it work. 

In fact, it should be other way round. Question one should be asking is  to create an endpoint - what data it would need and how one can normalise the underlying data structure so that it would be maintainable/scalable. 

I feel we are again introducing more tech debt which has to be paid ( with interest ). In any case the models in the API should not be modelled exactly.

Reply:
True, this spec is just one approach and might need some rethinking (especially the `/genomes` endpoint).

But let's step back for amoment. While we don't have to follow all the data models, we do need to:
 - stick to the track model (see `Track` schema object) in order for each track to have the data we expect
 - link each track object to a track catergory and genome id
 
In my view the main part in flux is the linking, either when submitting single or thousand tracks.
In this spec I've used the existing endpoints with a set of get/post/put methods to do the read/write operations at different granularity level (single track, groups of tracks). Implementing the spec in Django datamodels/serializers, the end result should look something like [this](https://services.ensembl-production.ebi.ac.uk/api/genome_metadata/) ([metadata api update service](https://services.ensembl-production.ebi.ac.uk/api/genome_metadata/)).

Reply:
The only purose for this endpoint in this form is to have genome IDs to validate against when adding first tracks for a new genome in `track_categories/{genome_id}` endpoint. I think we actually don't need it (see comments above).